src/scripts/main.js

```diff
@@ -1 +1,73 @@
 'use strict';
+
+const WAIT = 3000;
+const body = document.body;
+
+function createElemPromise(result) {
+  const elem = document.createElement('div');
+
+  elem.setAttribute('data-qa', 'notification');
+
+  if (result) {
+    elem.className = result;
+  }
+
+  return elem;
+}
+
+const elemFirstPromise = createElemPromise();
+const elemSecondPromise = createElemPromise('success');
+const elemThirdPromise = createElemPromise('success');
+
+const firstPromise = new Promise((resolve, reject) => {
+  function clickHandler() {
+    clearTimeout(timer);
+    resolve();
+  }
+
+  const timer = setTimeout(() => {
+    body.removeEventListener('click', clickHandler);
+    reject(new Error());
+  }, WAIT);
+
+  body.addEventListener('click', clickHandler);
+});
```

Comment:
The `firstPromise` logic is correct in terms of resolving or rejecting based on a click event or timeout. However, ensure that the `Error` object in the `reject` function on line 30 provides a meaningful message for better debugging.